ddl/ddl_worker.go

```diff
@@ -762,10 +764,14 @@ func (w *worker) HandleDDLJobTable(d *ddlCtx, job *model.Job) (int64, error) {
 	}
 	w.setDDLLabelForTopSQL(job.ID, job.Query)
 	w.setDDLSourceForDiagnosis(job.ID, job.Type)
-	jobContext := w.jobContext(job.ID)
+	jobContext := w.jobContext(job.ID, job.ReorgMeta)
 	if tagger := w.getResourceGroupTaggerForTopSQL(job.ID); tagger != nil {
 		txn.SetOption(kv.ResourceGroupTagger, tagger)
 	}
+	if job.ReorgMeta != nil {
+		txn.SetOption(kv.ResourceGroupName, job.ReorgMeta.ResourceGroupName)
+	}
```

Comment:
```suggestion
	txn.SetOption(kv.ResourceGroupName, jobContext.resourceGroupName)
```

How about using this? line767 does not `SetOption` @zimulala

Reply:
OK.

Reply:
fixed in https://github.com/pingcap/tidb/pull/46756/commits/37d96b3217f3a30e589c4d0060978487342e1cee